indexprovider/unsealedstatemanager.go

```diff
@@ -0,0 +1,272 @@
+package indexprovider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/filecoin-project/boost/db"
+	"github.com/filecoin-project/go-address"
+	cborutil "github.com/filecoin-project/go-cbor-util"
+	"github.com/filecoin-project/go-fil-markets/storagemarket"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/storage/sealer/storiface"
+	logging "github.com/ipfs/go-log/v2"
+	provider "github.com/ipni/index-provider"
+	"github.com/ipni/index-provider/metadata"
+	"time"
+)
+
+//go:generate go run github.com/golang/mock/mockgen -destination=./mock/mock.go -package=mock github.com/filecoin-project/go-fil-markets/storagemarket StorageProvider
+
+var usmlog = logging.Logger("unsmgr")
+
+type ApiStorageMiner interface {
+	StorageList(ctx context.Context) (map[storiface.ID][]storiface.Decl, error)
+}
+
+type UnsealedStateManager struct {
+	idxprov    *Wrapper
+	legacyProv storagemarket.StorageProvider
+	dealsDB    *db.DealsDB
+	sdb        *db.SectorStateDB
+	api        ApiStorageMiner
+}
+
+func NewUnsealedStateManager(idxprov *Wrapper, legacyProv storagemarket.StorageProvider, dealsDB *db.DealsDB, sdb *db.SectorStateDB, api ApiStorageMiner) *UnsealedStateManager {
+	return &UnsealedStateManager{
+		idxprov:    idxprov,
+		legacyProv: legacyProv,
+		dealsDB:    dealsDB,
+		sdb:        sdb,
+		api:        api,
+	}
+}
+
+func (m *UnsealedStateManager) Run(ctx context.Context) {
+	usmlog.Info("starting unsealed state manager")
+	ticker := time.NewTicker(time.Hour)
+	defer ticker.Stop()
+
+	// Check immediately
+	err := m.checkForUpdates(ctx)
+	if err != nil {
+		usmlog.Errorf("error checking for unsealed state updates: %s", err)
+	}
+
+	// Check every tick
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			err := m.checkForUpdates(ctx)
+			if err != nil {
+				usmlog.Errorf("error checking for unsealed state updates: %s", err)
+			}
```

Comment:
I think it would be good to wrap this with a timer, so that we build intuition on how long this process takes depending on how many sectors/deals there are in the database.